CONTRIBUTING.md

```diff
@@ -0,0 +1,167 @@
+
+- [Contributing to Contributivity](#contributing-to-contributivity)
+  - [1. Git workflow & branching](#1-git-workflow--branching)
+  - [2. Python](#2-python)
+    - [2.1 Python Virtual Environment](#21-python-virtual-environment)
+    - [2.2 Python Enhancement Proposals (PEP)](#22-python-enhancement-proposals-pep)
+      - [Black formatter](#black-formatter)
+        - [Installation](#installation)
+        - [Usage](#usage)
+      - [Flake8 linter](#flake8-linter)
+    - [2.3 Basic module structure & Imports order](#23-basic-module-structure--imports-order)
+    - [2.4 Jupyter Notebooks](#24-jupyter-notebooks)
+      - [2.5 Sharing & online rendering](#25-sharing--online-rendering)
+    - [3. urther Resources](#3-urther-resources)
+
+## 1. Git workflow & branching
+
+As long as other people contribute to a repository, it is easier and safer to restrain direct actions on the `master` branch. To do so, it is advised to create new branches, based on `master` (or a dedicated sub-branch, `dev` for example), to develop any new feature and then to open a Pull Request. Once audited and validated, it can be merged into `master`, and so on.
+
+To go further, here is a good example of a successful project build with a versioning file system (git, mercurial, etc.): <https://nvie.com/posts/a-successful-git-branching-model/>
+
+## 2. Python
+
+### 2.1 Python Virtual Environment
+
+In order to keep your installation of Substra separated from your general Python environement, which is a general Python good practice, it is recommanded to prepare a Python [virtual environment](https://virtualenv.pypa.io/en/latest/). In a new terminal window, please use one of the following method:
+
+```sh
+# Method 1: install the virtualenv package
+pip3 install --user virtualenv
+
+# Create a new virtual environment
+virtualenv -p python3 NAME_OF_YOUR_VENV
+# or even
+virtualenv -p $(which python3) NAME_OF_YOUR_VENV
+
+# Method 2: install the python3-venv package
+sudo apt install python3-venv # (Ubuntu)
+
+# Create a new virtual environment
+python3 -m venv NAME_OF_YOUR_VENV
+
+# Method 1 & 2: activate your new virtual env
+source NAME_OF_YOUR_VENV/bin/activate
+
+# Method 1 & 2: install Substra package inside your fresh new virtual environment
+pip3 install substra
+
+# Method 1 & 2: stop your virtual environment
+deactivate
+```
+
+Some of you might prefer to use [Anaconda](https://anaconda.org/), if so, you will be able to manage your virtual environment like this:
+
+```sh
+# Create a new virtual environment
+conda create --name NAME_OF_YOUR_VENV
+
+# Activate your new virtual environment
+conda activate NAME_OF_YOUR_VENV
+
+# Stop your virtual environment
+deactivate
+```
+
+Once inside your new virtual environment, you can install the project dependencies with the command `pip3 install -r requirements.txt`. What is installed inside a virtual environment is separated from your general Python setup.
```

Comment:
For conda: `conda install --file requirements.txt`

Reply:
Oh yes, thank you @RomainGoussault